### DISCUSSION TITLE
Better support for displaying and resolving Casted Enum

### DISCUSSION THREAD

**Discussion by sebastiaands** [laravel/nova-issues#4031] [Ideas & Feature Requests]
- Laravel Version: 9.8.1
- Nova Version: 4.2.2
- PHP Version: 8.1.4
- Database Driver & Version: MySQL 8
- Operating System and Version: MacOS 12.3.1
- Browser type and version: Google Chrome 100

### Description:
When casting a property to a PHP 8.1 enum (see https://laravel.com/docs/9.x/releases#enum-casting), a Nova Select field cannot be used. This leads to the following error:
```
exception: "InvalidArgumentException"
file: "/var/www/html/vendor/symfony/http-foundation/InputBag.php"
line: 70
message: "Excepted a scalar, or an array as a 2nd argument to \"Symfony\\Component\\HttpFoundation\\InputBag::set()\", \"App\\Enums\\PostType\" given."
```
This used to work in Laravel Nova 3.x.

### Detailed steps to reproduce the issue on a fresh Nova installation:
- Create a model with a property that casts to an enum
- Try to make this property editable in Nova using a Select field

**Comment by crynobone** [comment]
Please provide full reproducing repository.

**Comment by zek** [comment]
Same happens for Badge field.

**Comment by beebmx** [comment]
The problem is the expected value, and enums returns an object.
For the moment, and to fix it you can use this:

```php
use App\Enums\Status;
use Illuminate\Support\Collection;

Select::make('Status')
    ->options(
        Collection::make(Status::cases())->mapWithKeys(fn ($status) => [$status->value => $status->value])
    )->resolveUsing(fn($status) => $status->value
    )->displayUsing(fn($status) => $status->value)
```

The important points is `resolveUsing` and `displayUsing`

**Reply by KnudH** [reply]
I think this is not working anymore.
I tried exactly what you did, and I'm getting error: 
`Attempt to read property "value" on string"`

Ironically I though: okay cool, so let's use this string and get the Enum with from and then return the value
`MyEnum::from($status)->value` 

But this gives me error
`Object of class App\Enums\MyEnum could not be converted to string`

So its a string or its a enum? Nobody knows...

In the documentation of nova the world "enum" is not even mentioned. 1 year after php 8.1 with enums is released. That should be fixed in my opinion.

**Reply by crynobone** [reply]
If you have an issue or a bug submit a new "bug report" with reproducing code. 

FYI Laravel Nova 4 supports Enum partially because even if we support PHP 8.1 we still maintain support for PHP 7.3 and 7.4.
